src/pages/Staking/index.tsx

```diff
@@ -584,9 +590,18 @@ const Staking = (): JSX.Element => {
       return undefined;
     }
 
-    const remainingWeeksToUnstake = convertBlockNumbersToWeeks(remainingBlockNumbersToUnstake);
+    // If the user has staked
+    if (hasStakedAmount) {
+      if (remainingBlockNumbersToUnstake === null) {
+        throw new Error('Something went wrong!');
+      }
+      const remainingWeeksToUnstake = convertBlockNumbersToWeeks(remainingBlockNumbersToUnstake);
 
-    return Math.floor(STAKE_LOCK_TIME.MAX - remainingWeeksToUnstake);
+      return Math.floor(STAKE_LOCK_TIME.MAX - remainingWeeksToUnstake);
+    // If the user has not staked
+    } else {
+      return STAKE_LOCK_TIME.MAX;
+    }
```

Comment:
Similarly, the max extendable lock time is conditionally handled and properly returned.